bitcoin/src/blockdata/block.rs

```diff
@@ -98,14 +99,57 @@ impl Decodable for Version {
 /// * [CBlock definition](https://github.com/bitcoin/bitcoin/blob/345457b542b6a980ccfbc868af0970a6f91d1b82/src/primitives/block.h#L62)
 #[derive(PartialEq, Eq, Clone, Debug)]
 #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
-pub struct Block {
+pub struct Block<V = Unchecked>
+where
+    V: Validation,
+{
     /// The block header
     pub header: Header,
     /// List of transactions contained in the block
     pub txdata: Vec<Transaction>,
```

Comment:
In a867ba36b12b640bd65fdccac32e4796aab79d02:

These fields can't be public. Otherwise you can modify the txdata or header of a `Checked` block to be inconsistent.

Reply:
The `Checked` / `Unchecked` thing is just a marker, right? The aim is to try and help users do the right thing. I don't believe we can have the `Block<Checked>` type in a separate crate to the `Block<Unchecked>` type without a way to both construct the `Checked` one in `bitcoin` so there is no way to impose invariants.

Reply:
Said another way, currently we provide public `assume_checked` so there is no way to 100% enforce invariants.

Reply:
Public fields result in a "you can modify these fields without caring at all about invariants" API. That's totally different from an API where you can violate invariants but only by explicitly calling a method called `assume_checked`.

I'm not sure what you mean about separate crates.

Reply:
I must be learning, after I hit send I thought this is exactly what you would respond. I'll re-spin / re-think